unittest/test_costs_noFF.cpp

```diff
@@ -0,0 +1,258 @@
+  const boost::shared_ptr<crocoddyl::ActuationDataAbstract>& actuation_data = actuation->createData();
+  crocoddyl::DataCollectorActMultibody shared_data(&pinocchio_data, actuation_data);
+
+  // create the cost sum model
+  crocoddyl::CostModelSum cost_sum(state, model->get_nu());
+  cost_sum.addCost("myCost", model, 1.);
+
+  // Generating random values for the state and control
+  const Eigen::VectorXd& x = state->rand();
+
+  // Compute all the pinocchio function needed for the models.
+  crocoddyl::unittest::updateAllPinocchio(&pinocchio_model, &pinocchio_data, x);
+
+  BOOST_CHECK(model->get_state()->get_nx() == cost_sum.get_state()->get_nx());
+  BOOST_CHECK(model->get_state()->get_ndx() == cost_sum.get_state()->get_ndx());
+  BOOST_CHECK(model->get_nu() == cost_sum.get_nu());
+  BOOST_CHECK(model->get_state()->get_nq() == cost_sum.get_state()->get_nq());
+  BOOST_CHECK(model->get_state()->get_nv() == cost_sum.get_state()->get_nv());
+  BOOST_CHECK(model->get_activation()->get_nr() == cost_sum.get_nr());
+}
+
+void test_partial_derivatives_in_cost_sum(CostModelNoFFTypes::Type cost_type,
+                                          ActivationModelTypes::Type activation_type) {
+  // create the model
+  CostModelFactory factory;
+  const boost::shared_ptr<crocoddyl::CostModelAbstract>& model = factory.create(cost_type, activation_type);
+
+  // create the corresponding data object
+  const boost::shared_ptr<crocoddyl::StateMultibody>& state =
+      boost::static_pointer_cast<crocoddyl::StateMultibody>(model->get_state());
+  pinocchio::Model& pinocchio_model = *state->get_pinocchio().get();
+  pinocchio::Data pinocchio_data(pinocchio_model);
+
+  boost::shared_ptr<crocoddyl::ActuationModelAbstract> actuation =
+      boost::make_shared<crocoddyl::ActuationModelFull>(state);
+  const boost::shared_ptr<crocoddyl::ActuationDataAbstract>& actuation_data = actuation->createData();
+  crocoddyl::DataCollectorActMultibody shared_data(&pinocchio_data, actuation_data);
+  const boost::shared_ptr<crocoddyl::CostDataAbstract>& data = model->createData(&shared_data);
+
+  // create the cost sum model
+  crocoddyl::CostModelSum cost_sum(state, model->get_nu());
+  cost_sum.addCost("myCost", model, 1.);
+  const boost::shared_ptr<crocoddyl::CostDataSum>& data_sum = cost_sum.createData(&shared_data);
+
+  // Generating random values for the state and control
+  const Eigen::VectorXd& x = state->rand();
+  const Eigen::VectorXd& u = Eigen::VectorXd::Random(model->get_nu());
+
+  // Compute all the pinocchio function needed for the models.
+  crocoddyl::unittest::updateAllPinocchio(&pinocchio_model, &pinocchio_data, x);
+
+  // Computing the cost derivatives
+  model->calc(data, x, u);
+  model->calcDiff(data, x, u);
+
+  // Computing the cost-sum derivatives
+  cost_sum.calc(data_sum, x, u);
+  cost_sum.calcDiff(data_sum, x, u);
+
+  BOOST_CHECK((data->Lx - data_sum->Lx).isMuchSmallerThan(1.0));
```

Comment:
Is there is a reason to do not use the tolerance? (as in all the numdiff unit-tests)

Reply:
solved in #910